community/contributing/updating_the_class_reference.rst

```diff
@@ -154,220 +179,41 @@ repository:
-
-    func _ready():
-        var sprite = get_node("Sprite")
-        print(sprite.get_pos())
-
-If you need to have different code version in GDScript and C#, use ``[codeblocks]`` instead. If you use ``[codeblocks]``, you also need to have at least one of the language specific tags (``[gdscript]`` and ``[csharp]``). Always write GDScript code examples first! You can use this `experimental code translation tool <https://github.com/HaSa1002/codetranslator>`_ to speed up your workflow.
-
-.. code-block:: none
-
-    [codeblocks]
-    [gdscript]
-    func _ready():
-        var sprite = get_node("Sprite")
-        print(sprite.get_pos())
-    [/gdscript]
-    [csharp]
-    public override void _Ready()
-    {
-        var sprite = GetNode("Sprite");
-        GD.Print(sprite.GetPos());
-    }
-    [/csharp]
-    [/codeblocks]
-
-Will display as:
-
-.. tabs::
- .. code-tab:: gdscript GDScript
-
-    func _ready():
-        var sprite = get_node("Sprite")
-        print(sprite.get_pos())
-
- .. code-tab:: csharp
-
-    public override void _Ready()
-    {
-        var sprite = GetNode("Sprite");
-        GD.Print(sprite.GetPos());
-    }
+To learn to create a pull request, read `Creating a pull request
+<https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request>`_
+in the GitHub documentation.
 
+.. warning::
 
-To denote important information, add a paragraph starting with "[b]Note:[/b]" at
-the end of the description:
-
-.. code-block:: none
-
-    [b]Note:[/b] Only available when using the Vulkan renderer.
-
-To denote crucial information that could cause security issues or loss of data
-if not followed carefully, add a paragraph starting with "[b]Warning:[/b]" at the
-end of the description:
-
-.. code-block:: none
-
-    [b]Warning:[/b] If this property is set to [code]true[/code], it allows clients to execute arbitrary code on the server.
-
-For deprecated properties, add a paragraph starting with "[i]Deprecated.[/i]".
-Notice the use of italics instead of bold:
-
-.. code-block:: none
-
-    [i]Deprecated.[/i] This property has been replaced by [member other_property].
+    You should avoid editing files directly on GitHub in the godot engine
+    repository. As hundreds of contributors work on the engine, the Git history
+    must stay clean. Each commit should bundle all related improvements you make
+    to the class reference.
 
-In all the paragraphs described above, make sure the punctuation is part of the
-BBCode tags for consistency.
+    Editing from GitHub creates a new branch and Pull Request every time you
+    save your changes. Suppose days pass before your changes get a review. In
+    that case, you won't be able to update to the latest version of the
+    repository cleanly. Also, it's harder to keep clean indents with GitHub's
+    text editor. And they're essential in the class reference.
 
-I don't know what this method does!
+Updating the documentation template
```

Comment:
This whole section is partially outdated / needs to be reworked. We no longer recommend, nor *want*, doc contributors to create commits that "Sync the class reference".

Instead, we should describe how to use `doctool` to update/generate/validate XML templates when modifying the API of a class (as many doc writers are also engine contributors who document their changes). Changes unrelated to the documented class should be discarded with `git checkout`.

I guess I can do it myself once this is merged if you want.

Reply:
No worries, I took care of that.